src/execution_plan/ops/op_subquery_foreach.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright FalkorDB Ltd. 2023 - present
+ * Licensed under the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "RG.h"
+#include "op_argument.h"
+#include "op_argument_list.h"
+#include "op_subquery_foreach.h"
+#include "../execution_plan_build/execution_plan_util.h"
+
+// forward declarations
+static OpResult SubQueryForeachInit(OpBase *opBase);
+static Record SubQueryForeachConsume(OpBase *opBase);
+static void SubQueryForeachFree(OpBase *opBase);
+static OpBase *SubQueryForeachClone(const ExecutionPlan *plan, const OpBase *opBase);
+
+OpBase *NewSubQueryForeach
+(
+	const ExecutionPlan *plan
+) {
+	// validate inputs
+	ASSERT (plan != NULL) ;
+
+	OpSubQueryForeach *op = rm_calloc (1, sizeof (OpSubQueryForeach)) ;
+
+	// set operations
+	OpBase_Init ((OpBase *)op, OPType_SUBQUERY_FOREACH, "SubqueryForeach",
+			SubQueryForeachInit, SubQueryForeachConsume, NULL, NULL,
+			SubQueryForeachClone, SubQueryForeachFree, false, plan) ;
+
+	return (OpBase *)op ;
+}
+
+static OpResult SubQueryForeachInit
+(
+	OpBase *opBase
+) {
+	OpSubQueryForeach *op = (OpSubQueryForeach*)opBase;
+
+	ASSERT (OpBase_ChildCount (opBase) == 2) ;  // expecting 2 children
+
+	//--------------------------------------------------------------------------
+	// search for taps
+	//--------------------------------------------------------------------------
+
+	op->taps     = array_new (OpArgument*, 1) ;
+	OpBase **ops = array_new (OpBase*, 1) ;
+
+	OpBase *sub_query_root = OpBase_GetChild (opBase, 1) ;
+	array_append (ops, sub_query_root) ;
+
+	while (array_len (ops) > 0) {
+		OpBase *child = array_pop (ops) ;
+		OPType t = OpBase_Type (child) ;
+
+		// tap located
+		if ((OpBase_ChildCount (child) == 0) && (t == OPType_ARGUMENT)) {
+			array_append (op->taps, (OpArgument*) child) ;
+		}
+
+		// join op, traverse each branch
+		else if (t == OPType_JOIN || t == OPType_CARTESIAN_PRODUCT) {
```

Comment:
I see that is repeasted condition don't we want to add a constant array of join operations
or better register them as join operations so if another op is created we can register it once and all conditions will be updated